zerver/views/streams.py

```diff
@@ -380,6 +383,21 @@ def update_stream_backend(
     if stream_post_policy is not None:
         do_change_stream_post_policy(stream, stream_post_policy, acting_user=user_profile)
 
+    if can_remove_subscribers_group_id is not None:
+        if sub is None and stream.invite_only:
+            # Admins cannot change this setting for unsubscribed private streams.
+            raise JsonableError(_("Invalid stream ID"))
+
+        user_group = access_user_group_for_setting(
+            can_remove_subscribers_group_id,
+            user_profile,
+            setting_name="can_remove_subscribers_group",
+            allow_only_system_groups=True,
```

Comment:
`require_system_group=True` is probably clearer here?

And `cannot_set_to_internet_group` should probably be inverted; most permissions settings are going to be those that you need an account to use, so we should spell it `allow_internet_group=False`, which should probably be the default and thus not need to be spelled out here.

The other thought that I have here is that we probably are going to want to have this work more like the `property_types` framework. I think we could have a list in `models.py` with these settings, something like:
```
   stream_permission_group_settings = {
       "can_remote_subscribers_group": GroupPermissionSetting(
            require_system_group=True,
            ...
       ),
   }
```

and then this block eventually become a loop through those?

(I'm fine with this being a follow-up refactor if that makes sense, just trying to think through what design we'll want when we end up having like 10 stream settings that are based on groups).

Reply:
Updated the variables.

I was also thinking about how to write code when we will have many such settings, this idea sounds good. Would try to implement this next and will either add it to this PR or open a new PR.